compiler/bond.cabal

```diff
@@ -1,153 +1,166 @@
+    ld-options: -Wl,--dynamicbase -Wl,--nxcompat -Wl,--large-address-aware
   if os(windows) && arch(x86_64)
-    ld-options:     -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va
-
-  build-depends:    bond,
-                    aeson >= 0.7.0.6 && < 0.12.0.0,
-                    aeson-pretty == 0.7.2,
-                    base >= 4.9 && < 5,
-                    bytestring >= 0.10,
-                    cmdargs >= 0.10.10,
-                    directory >= 1.1,
-                    filepath >= 1.0,
-                    monad-loops >= 0.4,
-                    text >= 0.11,
-                    derive < 2.6,
-                    HUnit,
-                    QuickCheck,
-                    Diff >= 0.2 && < 0.4,
-                    pretty,
-                    tasty,
-                    tasty-golden,
-                    tasty-hunit,
-                    tasty-quickcheck,
-                    shakespeare >= 2.0,
-                    megaparsec >= 6.2 && < 7
+    ld-options: -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va
+  if flag(warningsAsError)
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N -Wall -Werror
+  else
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N
+  default-language: Haskell2010
 
 executable gbc
-  main-is:          Main.hs
-  other-modules:    IO
-                    Options
-
-  ghc-options:      -threaded
-  if flag(warningsAsErrors)
-    ghc-options:    -threaded -Wall -Werror
-
+  main-is: Main.hs
+  other-modules:
+      IO
+      Options
+      Paths_bond
+  hs-source-dirs:
+      ./.
+  build-depends:
+      aeson
+    , async
+    , base
+    , bond
+    , bytestring
+    , cmdargs
+    , directory
+    , filepath
+    , megaparsec
+    , monad-loops
+    , mtl
+    , process
+    , scientific
+    , shakespeare
+    , text
+    , unordered-containers
   if os(windows) && arch(i386)
-    ld-options:     -Wl,--dynamicbase -Wl,--nxcompat -Wl,--large-address-aware
+    ld-options: -Wl,--dynamicbase -Wl,--nxcompat -Wl,--large-address-aware
   if os(windows) && arch(x86_64)
-    ld-options:     -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va
+    ld-options: -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va
+  if flag(warningsAsError)
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N -Wall -Werror
+  else
+    ghc-options: -threaded -rtsopts -with-rtsopts=-N
+  default-language: Haskell2010
 
-  build-depends:    bond,
-                    aeson >= 0.7.0.6 && < 0.12.0.0,
-                    async >= 2.0.1.0,
-                    base >= 4.9 && < 5,
-                    bytestring >= 0.10,
-                    cmdargs >= 0.10.10,
-                    process < 1.7,
-                    directory >= 1.1,
-                    filepath >= 1.0,
-                    monad-loops >= 0.4,
-                    text >= 0.11,
-                    megaparsec >= 6.2 && < 7
+test-suite gbc-tests
+  type: exitcode-stdio-1.0
+  main-is: TestMain.hs
+  other-modules:
```

Comment:
Missing IO and Options according to the compiler:

```
Warning: The following modules should be added to exposed-modules or other-modules in C:\projects\bond\compiler\bond.cabal:
               - In test:gbc-tests:
                   IO
                   Options
               - In test:gbc-tests:
                   IO
                   Options
           
```

Reply:
Shall I move these two to be a part of the library? 
Both of these modules functions that can be very useful for plugin developers

Reply:
Hmm in my perspective we should just keep this PR to be a shift in the LTS and avoiding any regression. Pointed this out since I felt they were removed by mistake when moving things around see (plus compiler warning) https://github.com/microsoft/bond/pull/993/files#diff-8865cab0d973aeaba7290a70bfd3dcefL91

Reply:
Oops! I misuderstood the comment.
Fixed the warning.